### PR TITLE
INT-4443: Use SimpleEC for uriVariablesExpression

### PR DESCRIPTION
**Pull request by garyrussell**
JIRA: https://jira.spring.io/browse/INT-4443

__cherry-pick to 5.0.x, 4.3.x__ (what's new will need polishing)

### CODE REVIEW

File: build.gradle
Comment (artembilan) [inline build.gradle]: What version to use for `4.3.x` back-port ?

Thanks

**Review by artembilan — APPROVED** [review]
OK. Only one concern for back-porting to `4.3.x`.
However I agree that `changes-*.adoc` have to be reverted there. Not sure about `What's New` on the matter though...

**Review by artembilan — CHANGES_REQUESTED** [review]
Huh! Travis failed for many tests:
```
org.springframework.integration.http.outbound.HttpRequestExecutingMessageHandlerTests > testInt2455UriNotEncoded FAILED
    java.lang.IllegalArgumentException at HttpRequestExecutingMessageHandlerTests.java:761
```

File: build.gradle
Comment (garyrussell) [reply build.gradle]: https://github.com/spring-projects/spring-framework/commit/4bc3e0c2b4c703b3da90774e948f8adb18f334f9

4.3.15.BS

**Comment by garyrussell** [comment]
Strange; built ok for me.

**Comment by garyrussell** [comment]
Ah; new SPR commit: https://github.com/spring-projects/spring-framework/commit/38dcd179d5c2ad7630310cba292e80471b000d5f

File: spring-integration-core/src/main/java/org/springframework/integration/context/IntegrationContextUtils.java
Comment (artembilan) [inline spring-integration-core/src/main/java/org/springframework/integration/context/IntegrationContextUtils.java]: Must be `INTEGRATION_EVALUATION_CONTEXT_BEAN_NAME`.

Will fix on merge if only concern.

File: spring-integration-core/src/test/java/org/springframework/integration/configuration/EnableIntegrationTests.java
Comment (artembilan) [inline spring-integration-core/src/test/java/org/springframework/integration/configuration/EnableIntegrationTests.java]: Looks like the `this.context.getBean(StandardEvaluationContext.class);`
And this way we really ensure that `simple` variant is really not a `StandardEvaluationContext` impl.

Or do you have something else in mind on the matter ?

**Review by artembilan — APPROVED** [review]
Only two simple concerns.

I have them fixed (and some Copyrights) in hands: just need your approval and I'll merge this already.

Thanks

File: spring-integration-core/src/main/java/org/springframework/integration/context/IntegrationContextUtils.java
Comment (garyrussell) [reply spring-integration-core/src/main/java/org/springframework/integration/context/IntegrationContextUtils.java]: Oops; good catch.

File: spring-integration-core/src/test/java/org/springframework/integration/configuration/EnableIntegrationTests.java
Comment (garyrussell) [reply spring-integration-core/src/test/java/org/springframework/integration/configuration/EnableIntegrationTests.java]: No; that's good; I was just fixing the test because there are now 2 ECs.

**Comment by garyrussell** [comment]
Thanks; please go ahead and merge with your polishing.

**Comment by artembilan** [comment]
Back-port to `4.3.x` is blocked by the missed API in the SF-4.3.x.

Will come back to that when upstream is ready.

**Comment by artembilan** [comment]
Back-ported to `4.3.x` as https://github.com/spring-projects/spring-integration/commit/705d33a30c62d86774d7be9c2c606bdce3da08db

**Pull request closed without merging**
